docs/apache-airflow/extra-packages-ref.rst

```diff
@@ -343,27 +350,278 @@ you need to list explicitly all the non-bundle extras that you want to install.
 +---------------------+-----------------------------------------------------+------------------------------------------------------------------------+
 | all_dbs             | ``pip install 'apache-airflow[all_dbs]'``           | All database integrations                                              |
 +---------------------+-----------------------------------------------------+------------------------------------------------------------------------+
-| devel               | ``pip install 'apache-airflow[devel]'``             | Minimum development dependencies (without Hadoop, Kerberos, providers) |
-+---------------------+-----------------------------------------------------+------------------------------------------------------------------------+
-| devel_hadoop        | ``pip install 'apache-airflow[devel_hadoop]'``      | Adds Hadoop stack libraries to ``devel`` dependencies                  |
+
+Development extras
+------------------
+
+Generally none of the ``devel`` extras install providers - they expect the providers to be used from sources
+and those extras only make sense in editable mode. Users of Airflow should not be using them, unless they
+start contributing back and install airflow from sources.
+
+Those extras are only available in Airflow when it is installed in editable mode from sources
+(``pip install -e .``).
+
+Devel extras
+============
+
+The devel extras do not install dependencies for features of Airflow, but add functionality that is needed to
+develop Airflow, such as running tests, static checks. They do not install provider packages - even if they might be related
+to some providers (like ``devel_amazon``) but they might be needed if you want to test code of thoe corresponding
+provider.
+
+Even if some ``devel`` extras relate to providers - they do not install provider packages - for example
+``devel_amazon`` does not install amazon provider) but they might be needed if you want to test code of
+the corresponding provider (for example running mypy checks or running tests).
```

Comment:
paragraph has a text repetition of the paragraph before and a parenthesis closing missing an open. Trying to re-phrase what you wanted to say? Mhm, trying to re-phrase I assume it is just a redundant draft leftover which can be fully deleted?
```suggestion
```

Reply:
Removed now. yeah. It was repetitive. But with `devel_` removal it's even better because both repetitions are gone as we do not need to explain it any more... It should **just work** with regular extras.